packages/api-client-core/src/GadgetRecord.ts

```diff
@@ -98,7 +137,7 @@ export class GadgetRecordImplementation<Shape extends RecordShape> {
   changed(prop?: string | ChangeTracking, tracking = ChangeTracking.SinceLoaded) {
     return prop && typeof prop == "string"
       ? this.changes(prop, tracking).changed
-      : Object.keys(this.changes(prop === undefined ? tracking : (prop as ChangeTracking))).length > 0;
+      : this.hasChanges(prop === undefined ? tracking : (prop as ChangeTracking));
```

Comment:
@airhorns my implementation has a tradeoff of using more memory, but that cost seemed small, compared to the cost of:
1. Converting the fields object to an array.
2. Looping through all fields to generate a change object.
3. Converting that object to an array again.

I did a quick test of the `GadgetRecord.spec` test, and that was almost 10% faster. Considering the testing overhead would be the same for both, the performance diff should be even bigger.

Let me know if you don't think it's worth doing though, and I can revert this change.